src/smc-webapp/smc_chat.cjsx

```diff
@@ -323,27 +323,105 @@ Message = rclass
                 {cols}
             </Row>
 
+Array::bisect = (val) ->
```

Comment:
could you please move this block of code from smc_chat to smc-util/misc ? it looks very generic and could be useful anywhere. besides that, I'm not sure if it is so good to change the prototype. `bisect` is only used in one place, and there is nothing gained by doing this. Just define a normal function and call it `array_bisect`.

Also, once it is in misc.coffee, add some tests to `smc-util/test/misc-test.coffee`